### ISSUE TITLE
installed effect instances after effect aborted differ in backends

### ISSUE THREAD

**Issue by michaellilltokiwa** [tokiwa-software/fuzion#3676]
```
ex is

  b(a i32) : effect is


  abortable : effect is

  (b 3).instate_self ()->

    abortable.instate unit abortable (()->
      (b 4).instate_self ()->
        abortable.abort
    ) (()->unit)

    say b.env.a
```

JVM and Interpreter return: 3
C backend returns: 4

**Comment by fridis** [comment]
I am aware of this, I am currently revising the ugly code to return a result in case of an `abort`, but also the workaround using`EffectCall` in `instate0`, so this code will change in the backend.

**Comment by fridis** [comment]
This works now, with a slightly different example (added `_` for `default` lambda and added output of the inner value of `b.a` for info): 
```
ex is

  b(a i32) : effect is


  abortable : effect is

  (b 3).instate_self ()->

    abortable.instate unit abortable (()->
      (b 4).instate_self ()->
        say b.env.a
        abortable.abort
    ) (_->unit)

    say b.env.a
 > ./build/bin/fz -interpreter ex3676.fz
4
3
 > ./build/bin/fz -jvm ex3676.fz
4
3
 > ./build/bin/fz -c -o=ex3676 ex3676.fz && ./ex3676
4
3
```
This was fixed be #3784.
